week1/homework/out/Contact.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+exports.Contact = class Contact {
+	constructor(name, phone) {
+		this.name = name;
+		this.phone = phone;
+	}
+
+	get phone() {
+		return this._phone;
+	}
+	set phone(num) {
+		// convert string to int if it's valid number 
+		if (typeof num === 'string') {
+			// convert string to array
+			let a = num.split('');
+			// Make sure that( all )elements is integer 
+			let x = a.filter(function (i) {
+				return !isNaN(i);
+			}).length;
+			// If any element of this array not int (Unvalid number)
+			if (x !== a.length) {
+				console.log('Unvalid number');
+				return;
+			}
+			// If the strin is valid number convert to Int
+			else {
+					num = parseInt(num);
+				}
+		}
+
+		// validates that the number is  (non-negative,8 digits long, doesn't start with 0)
```

Comment:
how about writing separate variables for each of these three cases: 
non-negative,8 digits long, doesn't start with 0

and then instead write
```js
const nonNegative = ....;
const hasRightLength = ...;
const startsWithZero = ...;

if (nonNegative && hasRightLength && !startsWithZero) { ... }
```

don't think you think that's easier to read?

Reply:
Yes ,of course I will do that ,thank you :)